### PR TITLE
Fix motion cursor color for weapon

### PR DESCRIPTION
**Pull request by donnierussellii**
Also fix cursor-related crashes when changing screen resolution combined with entering/exiting cyberspace.

### CODE REVIEW

File: src/GameSrc/input.c [hunk 1]
```diff
@@ -1548,6 +1528,73 @@ uchar weapon_button_up = TRUE;
 // INTERNALS
 // ---------
 
+// -------------------------------------------------------------------------------------------
+// SetMotionCursorForMouseXY sets motion cursor for current mouse x,y position
+
+// This function is based view3d_mouse_input(); any changes there should be made here as well
```
Comment (Interrupt) [inline src/GameSrc/input.c]: Is there a way to have `view3d_mouse_input` call a common function to do this work without requiring it in two places?

Comment (donnierussellii) [reply src/GameSrc/input.c]: I'll try to consolidate them into one function.

**Comment by donnierussellii** [comment]
Added general-use CalcMotionCurOffset() that functions call.

File: src/GameSrc/fullscrn.c [hunk 2]
```diff
@@ -407,6 +407,9 @@ void fullscreen_start() {
     // KLC   uiShowMouse(NULL);
 
     CaptureMouse(true);
+
+    void SetMotionCursorForMouseXY(void);
```
Comment (Interrupt) [inline src/GameSrc/fullscrn.c]: Should these be `extern void` or does that not matter for these cases?

Comment (donnierussellii) [reply src/GameSrc/fullscrn.c]: Yes, making commits...

**Comment by Interrupt** [comment]
Ran into a crash testing this. Steps to reproduce:

- Start a new game
- Use the Ctrl+2 cheat to give yourself all weapons
- Switch to the Sparq beam
- Open the alcove door, and try to pick up the pipe

**Comment by donnierussellii** [comment]
I pick up the pipe and try to put it in the first slot, which should put slot one weapon on cursor. I think something wrong is going on in push_cursor_object().

This crash happens in the 0.6 release version as well, so I think it's unrelated to this PR.

I'll try to fix the crash with commits to this PR.

**Comment by donnierussellii** [comment]
The ctrl+2 cheat command adds weapons to an array, but does not actually create real obj's that can be dropped. When the game tries to put one of these "fake" objects on the cursor, it has nothing to work with.

Adding a null bmp check in push_cursor_object() to prevent the crash...

With a new game and the weapon slots all empty, when I ctrl+2 I can't drop any of the weapons.

But, starting with a new game, if I pick up the pipe, then ctrl+2, then right click the first weapon slot, the slot weapon disappears but nothing appears on the cursor.

The cheat commands in general are very hacky. For example, weapons are added without filling in all of the struct members, which can cause a crash if the ammo type is out of range.

**Pull request merged**